src/Rules/Properties/AccessStaticPropertiesRule.php

```diff
@@ -185,23 +185,27 @@ private function processSingleProperty(Scope $scope, StaticPropertyFetch $node,
 				return $messages;
 			}
 
-			if ($typeForDescribe instanceof TypeWithClassName && $typeForDescribe->getClassReflection() !== null) {
-				foreach ($typeForDescribe->getClassReflection()->getParents() as $parentClassReflection) {
-					if (!$parentClassReflection->hasProperty($name)) {
-						continue;
+			$classNames = $classType->getReferencedClasses();
```

Comment:
This is also wrong. `getReferencedClasses` will return nested classes too. Which means that if you get `Foo<Bar>`, getReferencedClasses will contain both Foo and Bar.

I think that in context of this PR, you need `TypeUtils::getDirectClassNames`. Feel free to fix the other PR too.

Reply:
at this point I just did what you said. I don't know what I am doing here right now.

> Feel free to fix the other PR too.

not sure which one you mean.

I will be in a meeting for the rest of the day. would be great in case you already know what needs to be done, if you could finish it.

thank you

Reply:
You said: refactored it to make it more like 
https://github.com/phpstan/phpstan-src/blob/582a9cb8b9b4fce2bd069bac26bf1d31dd52e7e2/src/Rules/Properties/AccessPropertiesRule.php#L119-L139

But that other rule is wrong too, not sure for how long.

Reply:
ok. I pushed a fix. hopefully thats what you had in mind.

thank you

Reply:
> But that other rule is wrong too, not sure for how long.

btw: I found the commit https://github.com/staabm/phpstan-src/commit/455fb7b890614c7f402fc1d87f7b2c26aa18424b